### DISCUSSION TITLE
Optimize NPM package manager with dedupe option

### DISCUSSION THREAD

**Discussion by jelhan** [renovatebot/renovate#25454] [Request Help]
### How are you running Renovate?

Self-hosted Renovate

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.33.5

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I am trying to get this working for the first time

### Describe the problem

NPM package manager with `npmDedupe` option is not very performant. I run into timeouts with my project. When looking into why it takes so much longer with deduplication, I noticed that the implementation could be improved.

Currently it does the following if `npmDedupe` is in `postUpdateOptions`:

1. Run `npm install` _without_ the `--package-lock-only` flag. This causes all packages to be downloaded and a `node_modules/` folder being created. Very time consuming. https://github.com/renovatebot/renovate/blob/3df10279cd3422c0ff815f37d8d6870105929462/lib/modules/manager/npm/post-update/npm.ts#L54-L63
2. Run `npm dedupe` as a second step. https://github.com/renovatebot/renovate/blob/3df10279cd3422c0ff815f37d8d6870105929462/lib/modules/manager/npm/post-update/npm.ts#L132-L136

Two optimizations are possible:

1. `npm install` can deduplicate the dependencies directly with the `--prefer-dedupe` flag. Instead of `npm install --no-audit && npm dedupe` we can do `npm install --prefer-dedupe`, which is much faster.
2. No need to download and install the packages for deduplication. `--prefer-dedupe` and `--package-lock-only` flags are working well together.

The performance difference is huge. `npm install --prefer-dedupe --package-lock-only --no-audit` is finished within 1 second on my test project while `npm install --no-audit && npm dedupe` takes minutes.

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

### Have you created a minimal reproduction repository?

Placeholder value, please select the correct response from the dropdown

**Comment by viceice** [comment]
please submit as feature request, as this isn't a wrong behavior in general.
